boards/arm/stm32h7/nucleo-h743zi/configs/mcuboot-app/defconfig

```diff
@@ -0,0 +1,142 @@
+#
+# This file is autogenerated: PLEASE DO NOT EDIT IT.
+#
+# You can use "make menuconfig" to make any modifications to the installed .config file.
+# You can then do "make savedefconfig" to generate a new defconfig file that includes your
+# modifications.
+#
+# CONFIG_SPI_EXCHANGE is not set
+# CONFIG_STM32H7_SYSCFG is not set
+CONFIG_APP_FORMAT_MCUBOOT=y
+CONFIG_APP_MCUBOOT_BOOTLOADER=y
```

Comment:
```suggestion
```

Reply:
I have some doubts respect this CONFIG flag. When selected it indicates the linker to use the `xxx-mcuboot-loader.ld` linker script in place of `xxx-mcuboot-app.ld`. This is confusing because of the word **APP** in the "CONFIG_APP_MCUBOOT_BOOTLOADER" flag. 

Has you think in using the inverse logic??
CONFIG_APP_MCUBOOT_BOOTLOADER=y     --> use xxx-mcuboot-app.ld
CONFIG_APP_MCUBOOT_BOOTLOADER=y     --> use xxx-mcuboot-laoder.ld

Reply:
Maybe reasonable to rename `xxx-mcuboot-app.ld` to `xxx-mcuboot-ota.ld`?
`CONFIG_APP_MCUBOOT_BOOTLOADER=y` should lead to picking `xxx-mcuboot-loader.ld` and `CONFIG_APP_MCUBOOT_BOOTLOADER=n` should pick other file. Probably  `xxx-mcuboot-ota.ld` name is better

Reply:
Or you can prepend `STM32H7` prefix, as an option

Reply:
Or just `STM32_` prefix like other option in same Kconfig like `STM32_ROMFS`

Reply:
Maybe also reasonable to remove `APP_` from `APP_FORMAT_MCUBOOT` and rename to `STM32_FORMAT_MCUBOOT`

Reply:
I like the option of renaming both APP_FORMAT_MCUBOOT and APP_MCUBOOT_BOOTLOADER to STM32_FORMAT_MCUBOOT and STM32_MCUBOOT_BOOTLOADER. 

Nevertheless as they are not specific configurations of STM32 boards I think they could be confusing in the future. Maybe it should be a standardized way of configuring the mcuboot for every boards and archs.

Reply:
Yes. I think STM32_FORMAT_MCUBOOT and STM32_MCUBOOT_BOOTLOADER are fine.
And yes, MCUboot options should be standardized in future across all the boards